src/coreclr/System.Private.CoreLib/src/System/Diagnostics/StackFrameHelper.cs

```diff
@@ -86,8 +86,40 @@ public StackFrameHelper()
         internal void InitializeSourceInfo(bool fNeedFileInfo, Exception? exception)
         {
             StackTrace.GetStackFramesInternal(this, fNeedFileInfo, exception);
```

Comment:
It might be nice to also defer Windows PDB data resolution, but that requires some nontrivial changes in the runtime.